### PR TITLE
Allow overriding core webpack plugins used by DynamicRemotePlugin

### PR DESCRIPTION
**Pull request by vojtechszocs**
This PR adds new `moduleFederationSettings.pluginOverride` option to webpack `DynamicRemotePlugin`.

This allows us to use libraries like [`@module-federation/enhanced`](https://github.com/module-federation/universe/tree/main/packages/enhanced) for better interoperability with tools like [Rspack](https://www.rspack.dev/).

For example:

```ts
import { ModuleFederationPlugin, ContainerPlugin } from '@module-federation/enhanced';

// in your webpack plugins declaration
new DynamicRemotePlugin({
  pluginMetadata,
  extensions,
  moduleFederationSettings: {
    // use webpack plugins from @module-federation/enhanced package (as imported above)
    pluginOverride: { ModuleFederationPlugin, ContainerPlugin }
  }
})
```

Related to https://github.com/scalprum/scaffolding/pull/127 - cc @Hyperkid123

### CODE REVIEW

**Review by Hyperkid123 — APPROVED** [review]

File: packages/lib-webpack/src/webpack/DynamicRemotePlugin.ts [hunk 1]
```diff
@@ -195,6 +216,11 @@ export class DynamicRemotePlugin implements WebpackPluginInstance {
     const moduleFederationLibraryType = moduleFederationSettings.libraryType ?? 'jsonp';
     const moduleFederationSharedScope = moduleFederationSettings.sharedScopeName ?? 'default';
 
+    const {
+      ModuleFederationPlugin = container.ModuleFederationPlugin,
+      ContainerPlugin = container.ContainerPlugin,
+    } = moduleFederationSettings.pluginOverride ?? {};
```
Comment (florkbr) [inline packages/lib-webpack/src/webpack/DynamicRemotePlugin.ts:222]: @vojtechszocs thoughts about writing this as:
```js
const {
  ModuleFederationPlugin,
  ContainerPlugin
} = moduleFederationSettings.pluginOverride || container;
```
?

Comment (vojtechszocs) [reply packages/lib-webpack/src/webpack/DynamicRemotePlugin.ts:222]: @florkbr I've checked the above code - it would result in both `ModuleFederationPlugin` and `ContainerPlugin` variables being possibly `undefined` so we'd still need the default assignments within that destructure statement, which then beats the `|| container` part :smiley: 

The reason for destructuring `moduleFederationSettings.pluginOverride ?? {}` instead of just `moduleFederationSettings.pluginOverride` is that the `pluginOverride` object itself can be `undefined` - TypeScript strict mode detects this as well, raising type errors if the `?? {}` part is missing.

TL;DR - I'd keep the original code.

JS destructuring assignment works with a single object getting picked from - imagine if `ModuleFederationPlugin` option was defined within the `pluginOverride` object but `ContainerPlugin` was not. In such case, the `|| container` part would have no effect.

Comment (florkbr) [reply packages/lib-webpack/src/webpack/DynamicRemotePlugin.ts:222]: Chatted with @vojtechszocs offline. My main concern here was readability - though we do have some limitations with object destructuring in JS. The alternative approach is not much clearer than what we have here - so I'm fine keeping as is for now.
